### PR TITLE
Special Event Plugin Interface

### PR DESCRIPTION
**Pull request by blamattina**
### Event Plugin Interface
```js
var tungsten = require('tungstenjs');
var specialPlugin = require('./my-awesome-plugin');
tungsten.addEventPlugin(specialPlugin);
```

### Changes
- Relocate special events from `src/event/handlers` to `plugins/`
- Add `npm run build-plugins` to build special event handlers
- Add `npm run clean` to clear `dist/`
- Update `npm run prepublish` to include `clean` and `build-plugins`
- Lay groundwork to move special event plugins to full fledged npm modules
- Added `plugin/README.md` with instructions on how to create a special event plugin
- Promoted `addEventPlugin` to public API
- Remove dependency on `underscore` for `intent`, `submit-data` and `touch`
- Update `browser-standalone` example to use a special event plugin

### TODO
- [x] Add documentation to demonstrate how these can be loaded [`docs/`?]
- [x] Squash

### CODE REVIEW

File: examples/browser-standalone/index.html [hunk 1]
```diff
@@ -10,29 +25,49 @@
 
 <script src="http://cdnjs.cloudflare.com/ajax/libs/underscore.js/1.8.3/underscore-min.js"></script>
 <script src="../../dist/tungsten.backbone.js"></script>
+
+<!-- tungsten event plugin -->
+<script src="../../dist/tungsten.event.intent.js"></script>
+
 <!-- Ractive.js for parsing templates -->
 <script src="http://cdn.ractivejs.org/latest/ractive.js"></script>
```
Comment (blamattina) [inline examples/browser-standalone/index.html]: @andrewrota Is this a current dependency?

File: package.json [hunk 2]
```diff
@@ -11,17 +11,19 @@
   "bugs": "https://github.com/wayfair/tungstenjs/issues",
   "main": "tungsten.js",
   "scripts": {
+    "clean": "rm dist/*.js",
```
Comment (blamattina) [inline package.json]: @andrewrota I'm not sure of the cross platform implications here.

File: src/event/global_events.js [hunk 3]
```diff
@@ -134,7 +127,7 @@ module.exports = {
   validateSelector: eventsCore.validateSelector,
   bindVirtualEvent: bindVirtualEvent,
   unbindVirtualEvent: unbindVirtualEvent,
-  registerEventHandler: registerEventPlugin,
+  registerEventHandler: registerEventHandler,
```
Comment (blamattina) [inline src/event/global_events.js]: This function had 3 internal names.  Now it has 2.  I'd like to standardize on 1... either `addEventPlugin` or `registerEventHandler`.

Any option @andrewrota @MattDeeg?

File: examples/browser-standalone/index.html (hunk 1 above)
Comment (andrewrota) [inline examples/browser-standalone/index.html]: Ractive isn't, no.

File: package.json (hunk 2 above)
Comment (andrewrota) [inline package.json]: It can probably go; I'm not sure where we use it.

Comment (blamattina) [inline package.json]: I would like to introduce `clean` as the first step in a build.  This implementation however is not travis friendly (and breaks ci).  I can pull this out for now or leave it in for utility.

Comment (andrewrota) [inline package.json]: Leave it for now (someone might be using it) and we can create a more compatible version later.

File: src/event/global_events.js (hunk 3 above)
Comment (andrewrota) [inline src/event/global_events.js]: I don't have a preference.

File: plugins/tungsten-event-touch/index.js [hunk 4]
```diff
@@ -157,14 +155,7 @@ if (typeof document.createEvent === 'function') {
...
-      'swipedown',
-      'swipeup',
-      'dbltap',
-      'tap'
-    ]
-  };
 }
+
+// Export an empty function to conform to the plugin API
+module.exports = function() {};
```
Comment (blamattina) [inline plugins/tungsten-event-touch/index.js]: @andrewrota Can you foresee any side effects we might encounter by returning an empty function here and registering it with `addEventPlugin`?  This keep the api uniform.

**Pull request merged**